openra/utility.py

```diff
@@ -11,14 +11,20 @@
 from subprocess import Popen, PIPE
 from django.conf import settings
 from django.utils import timezone
-from openra.models import Maps, Lints, MapCategories
+from openra.models import Maps, Lints, MapCategories, MapUpgradeLogs
 from openra import misc
 
 
 def map_upgrade(mapObject, engine, parser=list(reversed(list(settings.OPENRA_VERSIONS.values())))[0], new_rev_on_upgrade=True, upgrade_if_hash_matches=False, upgrade_if_lint_fails=False):
 
     parser_to_db = parser
+    parser_date = parser.split('-')[1]
     parser = os.path.join(settings.OPENRA_ROOT_PATH, parser)
+    engine_date = engine.split('-')[1]
+
+    # If parser date is newer than end of may 2018 use update-map and the full release name, otherwise use upgrade-map and the date only
+    # due to the syntax for this command changing in OpenRA
+    updated_openra_utility = int(parser_date) > 20180528
```

Comment:
The new commands have an additional constraint: `engine >= release-20171014`.  If someone wants to update from e.g. `release-20160508` to (whatever our next release ends up being) then they will first need to update to `release-20171014` using the the `release-20171014` engine and the old utility, and then to the latest version using the latest engine.

Would inserting something like the following here work?
```python
# HACK: we really shouldn't be assuming engine versions of the form <prefix>-<date>
if engine.split('-')[1] < '20171014':
    mapObject = map_upgrade(mapObject, engine, settings.OPENRA_VERSIONS['release-20171014'], new_rev_on_upgrade, upgrade_if_hash_matches, upgrade_if_lint_fails)
```

The alternative, which i've suggested elsewhere, is to delete all maps and parsers older than `release-20171014` and then support only the new-style updating.

Reply:
#337 removed the web-facing update feature for parsers < 20171014.  I plan to rebase, test, and merge this PR on top of these changes in the next day or two.